### PR TITLE
Refactor define-type parsing

### PR DESCRIPTION
**Pull request by eliaslfox**
The new dataflow is easier to follow.

### CODE REVIEW

**Review by gefjon — APPROVED** [review]
`parse-type-definitions` is still long as hell; any chance I can convince you to unload some parts of its logic into separate named functions?

**Comment by aijony** [comment]
> `parse-type-definitions` is still long as hell; any chance I can convince you to unload some parts of its logic into separate named functions?

I think a major point of this refactor was to do the exact opposite of this (as far as I can tell). Both methods have their merits, but in this case I think the sentimentality is more important than modularity.

File: src/typechecker/parse-type-definition.lisp [hunk 1]
```diff
@@ -289,128 +290,204 @@ Returns TYPE-DEFINITIONS"
...
+                            :runtime-type repr-arg
+                            :explicit-repr (list repr repr-arg)
+                            :enum-repr nil
+                            :newtype nil
+                            :constructors parsed-ctors
+                            :constructor-types ctor-types
+                            :docstring docstring))
+
+                          ((and newtype (eql repr :transparent))
+                           (let (;; The runtime type of a newtype is the runtime type of it's only constructor's only argument
```
Comment (aijony) [inline src/typechecker/parse-type-definition.lisp:370]: Probably should be "its", but eh.

**Review by aijony — APPROVED** [review]
I like this also.

**Comment by eliaslfox** [comment]
I think the data flow would be more difficult to track across multiple functions.

**Pull request merged**